app/Commands/NewCommand.php

```diff
@@ -0,0 +1,511 @@
+        $this->line($process->getOutput());
+
+        if ($this->replaceEnvVariables()) {
+            $this->info('I replaced .env variables in your new Laravel application');
+        }
+
+        if ($this->option('auth')) {
+            $this->doAuth();
+        }
+
+        if ($this->option('createdb')) {
+            $this->chooseDatabase($this->option('createdb'));
+        }
+
+        if ($this->option('node')) {
+            $this->doNodeOrYarn();
+        }
+
+        if ($this->hasTool('valet')) {
+            if ($this->option('link')) {
+                $this->doValetLink();
+            }
+            $this->openBrowser();
+        }
+
+        if ($this->hasTool('git')) {
+            $this->doGit();
+        }
+
+        $this->openTextEditor();
+
+        $this->info("You're ready to go! Remember to cd into '{$this->projectpath}' before you start editing.");
+    }
+
+
+    protected function setup()
+    {
+        $this->projectname = $this->argument('name');
+        $this->projecturl = 'http://' . $this->projectname . '.dev'; // @todo .dev is no longer good
+
+        $this->basepath = $this->cwd;
+        if ($this->option('path')) {
+            $path = $this->option('path');
+            if (is_dir($path)) {
+                $this->basepath = $path;
+            } else {
+                $this->warn("Your defined '--path $path' is not a directory, so I am skipping it and using '{$this->basepath}' instead.");
+            }
+        }
+
+        $this->projectpath = $this->basepath . DIRECTORY_SEPARATOR . $this->projectname;
+    }
+
+    protected function askToAndRemoveProject()
+    {
+        $this->info("The directory '{$this->projectpath}' already exists.");
+
+        if ($this->confirm("Shall I proceed by removing the following directory? {$this->projectpath}")) {
+            $fs = new Filesystem;
+
+            // @todo Use laravel --force here instead of deleteDirectory()?
+            if ($fs->deleteDirectory($this->projectpath)) {
+                $this->info("I removed the following directory: {$this->projectpath}");
+
+                return true;
+            } else {
+                $this->error("I was unable to remove the '{$this->projectpath}' directory so I must exit.");
+
+                return false;
+            }
+        } else {
+            $this->error("You have chosen to not remove the '{$this->projectpath}' directory so I must exit.");
+
+            return false;
+        }
+    }
+
+    protected function replaceEnvVariables()
+    {
+        // @todo make this a configuration option
+        $changes = [
+            'DB_DATABASE' => $this->projectname,
+            'DB_USERNAME' => 'root',
+            'DB_PASSWORD' => '',
+            'APP_URL'     => $this->projecturl,
+        ];
+
+        return $this->updateDotEnv($changes);
+    }
+
+    protected function doNodeOrYarn()
+    {
+        $command = null;
```

Comment:
Could we replace this with something like this?

```php
        if ($this->hasTool('yarn')) {
            $command = 'yarn';
        } else ($this->hasTool('npm')) {
            $command = 'npm install';
        } else {
            $this->error('Either yarn or npm are required');

            return false;
        } 
```

Reply:
Done, thanks